src/tainting/Taint.ml

```diff
@@ -255,7 +255,7 @@ module Taint_set = struct
      * coupling: If this changes, make sure to update docs for the `Taint.signature` type.
      *)
     set
-    |> Taint_map.update taint.orig (function
+    |> Taint_map.update taint (function
```

Comment:
And I don't think this is correct any more? This is not guaranteed to maintain `f(x) = x`.

Reply:
It works since the tests pass, but "breaks" what one would guess it's an invariant of the data structure.